### ISSUE TITLE
boost/1.72.0 bug

### ISSUE THREAD

**Issue by ghost** [conan-io/conan#9763]
### Environment Details (include every applicable attribute)
  * Operating System+version: window 10 pro
  * Compiler+version: gcc + 8
  * Conan version: 1.41
  * Python version:  3.8.8

------ by running this comman >conan install .. --build=missing

ERROR: Error, the file to download already exists: 'C:\Users\Lenovo\.conan\data\boost\1.72.0\_\_\dl\export\conan_sources.tgz'. [Remote: conancenter]
The 'boost/1.72.0' package has 'exports_sources' but sources not found in local cache.
Probably it was installed from a remote that is no longer available.

any clue on this???

**Comment by franramirez688** [comment]
Hi @botla04 

It usually happens whenever the remote from where you downloaded the package is not available anymore, then when you try to build that one locally again it fails.
I think, as a workaround, you should clean that `boost/1.72.0` package from your local cache and build it again, sorry.
We should improve that error message to explain better what's happening.

**Comment by ghost** [comment]
hello , 

i tried by deleting all data files in local cache in .conan/data folder
deleted binaries and packages , 
if i executing the conan install . --build=missing 
getting the below error .....


...failed updating 2 targets...
boost/1.72.0:
boost/1.72.0: ERROR: Package 'f0ee52ef9347335ea2fc37989d484b2921f9fd66' build failed
boost/1.72.0: WARN: Build folder C:\.conan\0506f3\1
ERROR: boost/1.72.0: Error in build() method, line 825
        self.run(full_command, run_environment=True)
        ConanException: Error 1 while executing b2.exe -q numa=on target-os=windows architecture=x86 address-model=64 binary-format=pe abi=ms --layout=system --user-config=C:\.conan\c30f49\1\source_subfolder\tools\build\user-config.jam -sNO_ZLIB=0 -sNO_BZIP2=0 -sNO_LZMA=1 -sNO_ZSTD=1 boost.locale.icu=off --disable-icu boost.locale.iconv=off --disable-iconv threading=multi visibility=hidden link=static variant=release --with-atomic --with-chrono --with-container --with-context --with-contract --with-coroutine --with-date_time --with-exception --with-fiber --with-filesystem --with-graph --with-iostreams --with-locale --with-log --with-math --with-program_options --with-random --with-regex --with-serialization --with-stacktrace --with-system --with-test --with-thread --with-timer --with-type_erasure --with-wave toolset=gcc define=_GLIBCXX_USE_CXX11_ABI=0 pch=on  cxxflags="-fPIC" install --prefix=C:\.conan\5f1926\1 -j8 --abbreviate-paths -d0 --debug-configuration --build-dir="C:\.conan\0506f3\1"


---------------------------------------------
more info :
default settings
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=8
compiler.libcxx=libstdc++
build_type=Release
[options]
[build_requires]
[env]

this is my conan remote:
conancenter: https://center.conan.io [Verify SSL: True]

conan version:
1.41.0

**Comment by memsharded** [comment]
This could be an issue of the ``boost`` recipe itself, that is not able to correctly compile in Windows with gcc, not a Conan issue.
You need to report more part of the terminal output, specifically that one including the actual build errors, not just the Conan message.

Any specific reason you are building with gcc in Windows? Typically packages from ConanCenter will have much better support for MSVC compiler, because that is the compiler it is used in the ConanCenter CI to build binaries by default, and MinGW support might be lacking or not enough in many recipes in the ConanCenter.

**Comment by dabele** [comment]
I get the same error message for boost in some circumstances. I'm not sure if it's the same cause.

So far, it only happens in windows github runners in our CI pipeline. The first run with a fresh .conan directory works fine. Then the second run fails with the same error message. The .conan directory (and nothing else) is cached between runs (I also tried only caching `.conan/data`, no change). The runs are only 30 mins apart, I don't know how the remote could have changed, I use the default anyway.

I'm using
* windows server 2019 github runner
* MSVC 2019
* conan 1.44.1, integrated into cmake.

packages:
```
        boost/1.75.0
        eigen/3.3.9
        spdlog/1.5.0
        jsoncpp/1.9.5
        hdf5/1.12.0
```

options 
```
        *:shared=False
        *:fPIC=True
        boost:without_atomic=True
        boost:without_chrono=True
        boost:without_container=True
        boost:without_context=True
        boost:without_contract=True
        boost:without_coroutine=True
        boost:without_date_time=True
        boost:without_exception=True
        boost:without_fiber=True
        boost:without_filesystem=False
        boost:without_graph=True
        boost:without_graph_parallel=True
        boost:without_iostreams=True
        boost:without_json=True
        boost:without_locale=True
        boost:without_log=True
        boost:without_math=True
        boost:without_mpi=True
        boost:without_nowide=True
        boost:without_program_options=True
        boost:without_python=True
        boost:without_random=True
        boost:without_regex=True
        boost:without_serialization=True
        boost:without_stacktrace=True
        boost:without_system=False
        boost:without_test=True
        boost:without_thread=True
        boost:without_timer=True
        boost:without_type_erasure=True
        boost:without_wave=True
        hdf5:enable_cxx=False
        hdf5:hl=False
```

<details><summary>Log of the first run (successful) </summary>

```
-- Conan: Detected VS runtime: MDd
-- Conan: checking conan executable
-- Conan: Found program C:/hostedtoolcache/windows/Python/3.7.9/x64/Scripts/conan.exe
-- Conan: Version found Conan version 1.44.1
-- Conan executing: C:/hostedtoolcache/windows/Python/3.7.9/x64/Scripts/conan.exe install . --remote conancenter --build missing --settings arch=x86_64 --settings build_type=Debug --settings compiler=Visual Studio --settings compiler.version=16 --settings compiler.runtime=MDd
Auto detecting your dev setup to initialize the default profile (C:\Users\runneradmin\.conan\profiles\default)
CC and CXX: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe, C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe 
No compiler was detected (one may not be needed)
Default settings
	os=Windows
	os_build=Windows
	arch=x86_64
	arch_build=x86_64
	build_type=Release
*** You can change them in C:\Users\runneradmin\.conan\profiles\default ***
*** Or override with -s compiler='other' -s ...s***


Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=Visual Studio
compiler.runtime=MDd
compiler.version=16
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

boost/1.75.0: Retrieving from server 'conancenter' 
boost/1.75.0: Trying with 'conancenter'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
boost/1.75.0: Downloaded recipe revision 0
eigen/3.3.9: Retrieving from server 'conancenter' 
eigen/3.3.9: Trying with 'conancenter'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
eigen/3.3.9: Downloaded recipe revision 0
spdlog/1.5.0: Retrieving from server 'conancenter' 
spdlog/1.5.0: Trying with 'conancenter'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
spdlog/1.5.0: Downloaded recipe revision 0
fmt/6.2.1: Retrieving from server 'conancenter' 
fmt/6.2.1: Trying with 'conancenter'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
fmt/6.2.1: Downloaded recipe revision 0
jsoncpp/1.9.5: Retrieving from server 'conancenter' 
jsoncpp/1.9.5: Trying with 'conancenter'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
jsoncpp/1.9.5: Downloaded recipe revision 0
hdf5/1.12.0: Retrieving from server 'conancenter' 
hdf5/1.12.0: Trying with 'conancenter'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
hdf5/1.12.0: Downloaded recipe revision 0
zlib/1.2.11: Retrieving from server 'conancenter' 
zlib/1.2.11: Trying with 'conancenter'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
zlib/1.2.11: Downloaded recipe revision 0
b2/4.5.0: Retrieving from server 'conancenter' 
b2/4.5.0: Trying with 'conancenter'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
b2/4.5.0: Downloaded recipe revision 0
conanfile.txt: Installing package
Requirements
    boost/1.75.0 from 'conancenter' - Downloaded
    eigen/3.3.9 from 'conancenter' - Downloaded
    fmt/6.2.1 from 'conancenter' - Downloaded
    hdf5/1.12.0 from 'conancenter' - Downloaded
    jsoncpp/1.9.5 from 'conancenter' - Downloaded
    spdlog/1.5.0 from 'conancenter' - Downloaded
    zlib/1.2.11 from 'conancenter' - Downloaded
Packages
    boost/1.75.0:7621b53e2cefb7a76565912308e57d14463b92df - Build
    eigen/3.3.9:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Download
    fmt/6.2.1:d057732059ea44a47760900cb5e4855d2bea8714 - Download
    hdf5/1.12.0:487a5ebc62be09517ca7a9dbf8f7f4a1f219019b - Build
    jsoncpp/1.9.5:d057732059ea44a47760900cb5e4855d2bea8714 - Download
    spdlog/1.5.0:a39a6b0bf7a79964cffa12efbc32e1a7db2952c7 - Download
    zlib/1.2.11:d057732059ea44a47760900cb5e4855d2bea8714 - Download
Build requirements
    b2/4.5.0 from 'conancenter' - Downloaded
Build requirements packages
    b2/4.5.0:ca33edce272a279b24f87dc0d4cf5bbdcffbc187 - Download
...
...
```

</details>

<details><summary>Log of the second run (error)</summary>

```
-- Conan: Detected VS runtime: MDd
-- Conan: checking conan executable
-- Conan: Found program C:/hostedtoolcache/windows/Python/3.7.9/x64/Scripts/conan.exe
-- Conan: Version found Conan version 1.44.1
-- Conan executing: C:/hostedtoolcache/windows/Python/3.7.9/x64/Scripts/conan.exe install . --remote conancenter --build missing --settings arch=x86_64 --settings build_type=Debug --settings compiler=Visual Studio --settings compiler.version=16 --settings compiler.runtime=MDd
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=Visual Studio
compiler.runtime=MDd
compiler.version=16
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

ERROR: Error, the file to download already exists: 'C:\Users\runneradmin\.conan\data\boost\1.75.0\_\_\dl\export\conan_sources.tgz'. [Remote: conancenter]
The 'boost/1.75.0' package has 'exports_sources' but sources not found in local cache.
Probably it was installed from a remote that is no longer available.

conanfile.txt: Installing package
Requirements
    boost/1.75.0 from 'conancenter' - Cache
    eigen/3.3.9 from 'conancenter' - Cache
    fmt/6.2.1 from 'conancenter' - Cache
    hdf5/1.12.0 from 'conancenter' - Cache
    jsoncpp/1.9.5 from 'conancenter' - Cache
    spdlog/1.5.0 from 'conancenter' - Cache
    zlib/1.2.11 from 'conancenter' - Cache
Packages
    boost/1.75.0:7621b53e2cefb7a76565912308e57d14463b92df - Build
    eigen/3.3.9:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    fmt/6.2.1:d057732059ea44a47760900cb5e4855d2bea8714 - Cache
    hdf5/1.12.0:487a5ebc62be09517ca7a9dbf8f7f4a1f219019b - Cache
    jsoncpp/1.9.5:d057732059ea44a47760900cb5e4855d2bea8714 - Cache
    spdlog/1.5.0:a39a6b0bf7a79964cffa12efbc32e1a7db2952c7 - Cache
    zlib/1.2.11:d057732059ea44a47760900cb5e4855d2bea8714 - Cache
Build requirements
    b2/4.5.0 from 'conancenter' - Cache
Build requirements packages
    b2/4.5.0:ca33edce272a279b24f87dc0d4cf5bbdcffbc187 - Cache

Installing (downloading, building) binaries...
b2/4.5.0: Already installed!
eigen/3.3.9: Already installed!
fmt/6.2.1: Already installed!
jsoncpp/1.9.5: Already installed!
zlib/1.2.11: Already installed!
boost/1.75.0: Applying build-requirement: b2/4.5.0
Downloading conan_sources.tgz
CMake Error at cmake/conan.cmake:638 (message):
  Conan install failed='1'
Call Stack (most recent call first):
  cmake/memilio-conan.cmake:38 (conan_cmake_install)
  thirdparty/CMakeLists.txt:5 (memilio_conan_install)
  CMakeLists.txt:71 (include)
```
</details>

Disabling the cache helps of course, but increases build times. I don't have that problem on my windows laptop, it can reuse packages from .conan without issue. Any ideas?

**Comment by dabele** [comment]
I found the root of my issue above. I did not know about the short paths workaround on windows. In addition to the conan user home (e.g. ~/.conan) the conan user_home_short needs to be cached between runs. Not sure why I missed the short paths issue in the documentation, maybe it needs to be highlighted a bit better? The error message could also be more clear here.

Anyway, my report above is resolved.
